### PR TITLE
Add selected README development guidelines and references for loopps

### PR DESCRIPTION
**Pull request by krehermann**

### CODE REVIEW

File: pkg/loop/internal/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- create a `test` package (`chainlink-common/pkg/loop/internal/core/services/mynewservice/test`) that provides a static implementation of your interface. see existing code for examples of the pattern, and documentation in `pkg/loop/internal/test`
+
+Modifications follow a very similar pattern: update the interface (in needed), update the proto to reflect any interface changes, implement and test the changes. See Protobuf best practices[[Protobuf best practices]]
+for details ensuring backward compatibility of protobufs.
+
+### Relayer abstraction Development
+
+There are three kinds of abstractions with the relayer: common components, such as the `chainreader`, ocr-specific shared components, such as `offchain digester` and the ocr-specific, product-specific components, known as `Providers`. Note that the line between the latter to categories is can be blurry, ocr-specific shared component should be thought as component that have a well defined and reasonable default value; custom `Providers` clearly fall outside this domain, but custom `Providers` may also implement OCR components that have a default implementation.
```
Comment (cedric-cordenier) [inline pkg/loop/internal/README.md:52]: typo: latter to -> latter two

**Review by cedric-cordenier — DISMISSED** [review]

File: pkg/loop/internal/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+There are roughly two kinds of development, adding entirely new services and modifying existing services.
+
+The flow to add a new service is 
+- define your interface in the `chainlink-common/pkg/types` tree 
+- create the proto definition in a sub package of `pb`. 
+    - golang `interface` maps to proto `service` and `struct` map to `message` (see the `pb` README for details about organization and code gen)
+- create a pkg in `chainlink-common/pkg/loop/internal/core/services` and implement the gRPC server and client
+- create a `test` package (`chainlink-common/pkg/loop/internal/core/services/mynewservice/test`) that provides a static implementation of your interface. see existing code for examples of the pattern, and documentation in `pkg/loop/internal/test`
+
+Modifications follow a very similar pattern: update the interface (in needed), update the proto to reflect any interface changes, implement and test the changes. See Protobuf best practices[[Protobuf best practices]]
```
Comment (george-dorin) [inline pkg/loop/internal/README.md:47]:
```suggestion
Modifications follow a very similar pattern: update the interface (if needed), update the proto to reflect any interface changes, implement and test the changes. See Protobuf best practices[[Protobuf best practices]]
```

File: pkg/loop/internal/README.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+Developing common relayer components such as `chainreader` follows the same pattern as [Core node abstraction Development]. The primary difference is the package location. Instead of adding code to `chainlink-common/pkg/loop/internal/core/services` you will work in `chainlink-common/pkg/loop/internal/relayer` or `chainlink-common/pkg/loop/internal/relayer/pluginprovider` depending on the scope of your changes.
+
+### Product abstraction Development
+
+Product specific development is the most nuanced kind of development. There are at least three different paths:
+
+1. Use the Core Node API, without a custom provider.
+
+    This is vision, but will not be viable until chain reading and writing abstractions are implemented.
```
Comment (george-dorin) [inline pkg/loop/internal/README.md:64]:
```suggestion
    This is our vision, but will not be viable until chain reading and writing abstractions are implemented.
```

File: pkg/loop/internal/relayer/pluginprovider/ext/ccip/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+1. Update the golang interfaces in `chainlink-common/pkg/types`, and/or  `chainlink-common/pkg/types/ccip`
+2. Mirror those changes as protobuf changes in `chainlink-common/pkg/loop/internal/pb/ccip`
+    note: golang `interface` <-> gRPC `service` and `struct` <-> `message`
+    remember to run `make generate` at the top of the repo to generate your changes
+    see docs in `internal/pb` for more details
+3. If you are adding a new interface, create a file here and implement the gRPC server and client. Follow the testing pattern in the  `test` subdir to ensure roundtrip de/serialization of each and every rpc
+4. If you are updating an interface, then update both the `GRPCClient` and `GRPCServer` in the relevant file.
+    - The client code will not compile if you change the interface in `chainlink-common/pkg/types/ccip` and so these are obvious
+    - However the server code will compile and you will receive runtime `Unimplemented` errors (this is a consequence of golang gRPC best practices that embed a `UnimplementedServer`. the curious reader can go here)
```
Comment (george-dorin) [inline pkg/loop/internal/relayer/pluginprovider/ext/ccip/README.md:13]: For debug you can comment out the `UnimplementedServer` to see what methods are missing/broken

**Review by george-dorin — APPROVED** [review]

**Pull request merged**